src/main/scala/sbtcrossproject/CrossProject.scala

```diff
@@ -1,184 +1,115 @@
 package sbtcrossproject
 
-import scala.language.implicitConversions
-import scala.language.experimental.macros
-
-import scala.reflect.macros.Context
-
-import sbt._
-import Keys._
-import Project.projectToRef
-
 import java.io.File
 
-final class CrossProject private (
-    crossType: CrossType,
-    val projects: Map[CrossPlatform, Project]
-) {
-  import CrossProject._
-
-  // Transformers for inner projects
-
-  def configure(platform: CrossPlatform, transform: Project => Project) =
-    copy(projects = projects + (platform -> transform(projects(platform))))
-
-  def settings(platform: CrossPlatform, ss: Def.Setting[_]*): CrossProject =
-    configure(platform, _.settings(ss: _*))
-
-  // Scala.js-style cross project aliases
-
-  /** Project for JVM platform. */
-  def jvm: Project = projects(JVM)
-
-  /** Transform the underlying JVM project */
-  def jvmConfigure(transform: Project => Project): CrossProject =
-    configure(JVM, transform)
+import sbt.{AutoPlugin, Configuration, Plugins, Project, ProjectReference}
 
-  /** Add settings specific to the underlying JVM project */
-  def jvmSettings(ss: Def.Setting[_]*): CrossProject =
-    settings(JVM, ss: _*)
-
-  // Concrete alteration members
-
-  def aggregate(refs: CrossProject*): CrossProject =
-    mapCrossPlatforms { p =>
-      projects(p).aggregate(refs.map(_.projects(p): ProjectReference): _*)
-    }
-
-  def configs(cs: Configuration*): CrossProject =
-    mapProjects(_.configs(cs: _*))
+import scala.language.experimental.macros
+import scala.language.implicitConversions
+import scala.reflect.macros.Context
 
-  def configureCross(transforms: (CrossProject => CrossProject)*): CrossProject =
-    transforms.foldLeft(this)((p, t) => t(p))
+/**
+ * Created by grinder on 17.07.16.
+ */
+class CrossProject[A <: Platform] private (
+    id: String,
+    root: File,
+    val projects: Map[Platform, Project]) {
+  def settings(t: Platform,
+               settings: Seq[_root_.sbt.Def.Setting[_]]): CrossProject[A] =
+    copy(projects + (t -> projects(t).settings(settings)))
+
+  def path(t: Platform, path: String): CrossProject[A] =
+    copy(projects + (t -> projects(t).in(new File(s"$root/$path"))))
+
+  def platform[U <: Platform](t: U): CrossProject[A with U] = {
```

Comment:
I think it is dangerous, from an API point of view, to allow `.platform` to be called after things that `.settings`. Users will have wrong expectations on the outcome of such code.

It would be safer to have a `CrossProjectBuilder` that has only a `.platform` method, and which can be implicitly converted to `CrossProject`. `CrossProject` does not have `.platform`. That way you can chain several calls to `.platform`, then several calls `.settings`, but not call `.platform` again after having called `.settings`.